src/service/chat.js

```diff
@@ -232,9 +231,19 @@ const initChatFromPost = async ({source, id}) => {
   }
 };
 
-const getAllowAnonDmStatus = async (postId) => {
+const getAllowAnonDmStatus = async (type, id) => {
+  const generateUrl = () => {
+    switch (type) {
+      case 'post':
+        return `/users/check-allow-dm/?source=post&post_id=${id}`;
+      case 'comment':
+        return `/users/check-allow-dm/?source=comment&comment_id=${id}`;
+      default:
+        return '';
+    }
+  };
   try {
-    const response = await api.get(`/users/check-allow-dm/?source=post&post_id=${postId}`);
+    const response = await api.get(generateUrl());
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The refactoring done here to generate the URL based on the `type` parameter is a good approach. It makes the function more flexible and easier to maintain. However, similar to the previous hunk, there should be error handling for the case when `type` is neither 'post' nor 'comment'.

```diff
+    default:
+      throw new Error(`Invalid type: ${type}`);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->